tools/SetupFlow/DevHome.SetupFlow/ViewModels/Environments/EnvironmentCreationOptionsViewModel.cs

```diff
@@ -153,21 +172,45 @@ protected async override Task OnEachNavigateToAsync()
         CanGoToNextPage = false;
         var curSelectedProviderId = _curProviderDetails?.ComputeSystemProvider?.Id ?? string.Empty;
         var upcomingSelectedProviderId = _upcomingProviderDetails?.ComputeSystemProvider?.Id;
-        _isFirstTimeLoadingNewFlow = true;
 
         // Selected compute system provider changed so we need to update the adaptive card in the UI
         // with a new adaptive card from the new provider.
         _curProviderDetails = _upcomingProviderDetails;
-
-        IsAdaptiveCardSessionLoaded = false;
+        SelectedDeveloperId = new DeveloperIdWrapper(new EmptyDeveloperId());
         AdaptiveCardLoadingMessage = StringResource.GetLocalized(StringResourceKey.EnvironmentCreationUILoadingMessage, _curProviderDetails.ComputeSystemProvider.DisplayName);
+        DeveloperIdWrappers.Clear();
+        CreationPageState = CreationPageStateKind.AccountSelection;
+
+        // The DeveloperId count will always be at least one for every provider. If there is only one,
```

Comment:
You sure?  What is preventing a user making an environment and them not logging in?  Does the UI prevent this scenario?

Reply:
the way how environments in Dev Home works is that we query the compute system providers for their compute systems using a Dev ID. To do this we first get all the DevIds from the extension, and then for each one we query the compute system provider for compute systems related with the Dev Id.

If there are no DevIds associated with an extension we artificially create an empty one for that provider. See here: when there is only one developerId, that means that it is the empty developerId here: https://github.com/microsoft/devhome/blob/9654c0e647d60a37ec7f1df8f7120d06c50d60d7/common/Services/ComputeSystemService.cs#L80. We then send that to the extensions when querying for compute systems. Some extensions do need devIds but some do not and we only have one SDK method to retrieve them which takes a DevId. So, in those cases it is expected that the extension ignores the empty DevId and sends us their compute systems. E.g Hyperv and wsl extensions are like this.

So this means that even if there are no DevIds from a provider we're guaranteed to always have exactly one in this call. For the case  where the user hasn't logged in yet, we'd send the empty one and the expectation is that the extension returns a failure SDK result (assuming it actually needs a DevId e.g the Dev box provider).

The user would need to leave the page, login through Dev Home settings, come back to this page and click the previous button then the next button again. Then they will see UI for the compute system provider, or the dropdown if they logged in two separate accounts.

Reply:
I see.  Thank you for the explanation.  Maybe adding a comment about why the count is always 1 would help future developers.  Something like `If a user is not logged in, then environments makes an empty DeveloperId`